app/helpers/application_helper/toolbar_builder.rb

```diff
@@ -796,7 +796,6 @@ def build_toolbar_hide_button(id)
       when "miq_template_refresh"
         return true if @record && !@record.ext_management_system && !(@record.host && @record.host.vmm_product.downcase == "workstation")
       when "miq_template_scan", "image_scan"
-        return true unless @record.is_available?(:smartstate_analysis) || @record.is_available_now_error_message(:smartstate_analysis)
```

Comment:
As I said in earlier comment, this line never gets executed because the OR condition always returns true. I still have to confirm from @h-kataria whether it should depend on archived? value, if yes, then archived? condition should be added here because only in that case, OR condition gave false value for smartstate_analysis.